samcli/commands/init/init_templates.py

```diff
@@ -112,14 +105,23 @@ def _check_app_template(entry: Dict, app_template: str) -> bool:
         return bool(entry["appTemplate"] == app_template)
 
     def init_options(self, package_type, runtime, base_image, dependency_manager):
-        if not self.clone_attempted:
-            self._clone_repo()
-        if self.repo_path is None:
+        if not self._git_repo.clone_attempted:
+            shared_dir: Path = global_cfg.config_dir
+            try:
+                self._git_repo.clone(clone_dir=shared_dir, replace_existing=True)
+            except CloneRepoUnstableStateException as ex:
+                raise AppTemplateUpdateException(str(ex)) from ex
+            except (OSError, CloneRepoException):
```

Comment:
`CloneRepoException` can be also raised if `replace_existing=False` and `Path(dest_path).exists()`, which does not seem to match the error handling behavior here.

Reply:
this call has replace_existing set to True

Reply:
Oh I see, when is `replace_existing` set to `False`?

Reply:
GitRepo doesn't care about this. it is a case by case thing